datasets/switchboard.py

```diff
@@ -336,3 +339,325 @@ def run(self):
                 token = parts[0].replace("&amp;", "&")  # e.g A&amp;E -> A&E
                 mapped_token = _map_token(token)  # preprocessing as corpus
                 out_f.write(mapped_token + " " + parts[1] + "\n")
+
+
+class SwitchboardSphereToWave(Job):
+    """
+    Takes an audio folder from one of the switchboard LDC folders and converts dual channel .sph files
+    with mulaw encoding to single channel .wav files with s16le encoding
+    """
+
+    def __init__(self, sph_audio_folder: tk.Path):
+        """
+        :param sph_audio_folder:
+        """
+        self.sph_audio_folder = sph_audio_folder
+
+        self.out_wave_audio_folder = self.output_path("wave_audio", directory=True)
+
+        self.rqmt = {"cpu": 1, "mem": 1, "time": 1.0}
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        for sph_file in glob.glob(
+            os.path.join(self.sph_audio_folder.get_path(), "**/*.sph"), recursive=True
+        ):
+            sph_name, ext = os.path.splitext(os.path.basename(sph_file))
+            subprocess.call(
+                [
+                    "ffmpeg",
```

Comment:
why not `sox`? Do we know what differs here?
I think Zoltan's original script was using `sox` and also AppTek for `raw->wav` generally uses it.

Reply:
I do not want to add an additional dependency to the pipeline, everything else works with FFmpeg.
I am more familiar with FFmpeg, and the results do not (or rather, can not) differ (Peter and I checked), as this is numerically stable conversion from mulaw-pcm to linear-pcm.